### PR TITLE
Add ReportsController spec

### PR DESCRIPTION
**Pull request by eric-kovshovik**
#29 Adding spec for `ReportsController`

### CODE REVIEW

**Comment by Daibushi** [comment]
Great work guy.

File: spec/controllers/api/v1/reports_controller_spec.rb [hunk 1]
```diff
@@ -1,5 +1,43 @@
 require 'rails_helper'
 
 RSpec.describe Api::V1::ReportsController, type: :controller do
+  login_user
 
+  describe '#create' do
+    let(:user) { User.new(username: 'user1') }
```
Comment (AlexAvlonitis) [inline spec/controllers/api/v1/reports_controller_spec.rb]: Could you change this to a factory please: 
```
let(:user) { FactoryBot.build(:user) }
```

File: spec/controllers/api/v1/reports_controller_spec.rb [hunk 2]
```diff
@@ -1,5 +1,43 @@
...
+      end
+    end
+
+    context 'when user was found' do
+      before do
+        allow(User).to receive(:find_by!).and_return(user)
+      end
+
+      it "returns json report" do
+        post :create, format: :json, params: { username: user.username, reason: 'γιατί όχι', description: 'ἐχω χἀσει τα αυγἀ και τα καλἀθια' }
```
Comment (AlexAvlonitis) [inline spec/controllers/api/v1/reports_controller_spec.rb]: that's great :D

**Comment by AlexAvlonitis** [comment]
Great job @eric-kovshovik added a minor comment

**Pull request merged**